app/src/main/java/com/keylesspalace/tusky/fragment/ViewImageFragment.kt

```diff
@@ -181,99 +201,18 @@ class ViewImageFragment : ViewMediaFragment() {
     }
 
     override fun onDestroyView() {
-        Glide.with(this).clear(photoView)
-        transition.onComplete()
         super.onDestroyView()
+        photoView.ssiv?.recycle()
     }
 
-    private fun loadImageFromNetwork(url: String, previewUrl: String?, photoView: ImageView) {
-        val glide = Glide.with(this)
-        // Request image from the any cache
-        glide
-                .load(url)
-                .dontAnimate()
-                .onlyRetrieveFromCache(true)
-                .let {
-                    if (previewUrl != null)
-                        it.thumbnail(glide
-                                .load(previewUrl)
-                                .dontAnimate()
-                                .onlyRetrieveFromCache(true)
-                                .addListener(ImageRequestListener(true, isThumnailRequest = true)))
-                    else it
-                }
-                //Request image from the network on fail load image from cache
-                .error(glide.load(url)
-                        .centerInside()
-                        .addListener(ImageRequestListener(false, isThumnailRequest = false))
-                )
-                .addListener(ImageRequestListener(true, isThumnailRequest = false))
-                .into(photoView)
-    }
-
-    /**
-     * We start transition as soon as we think reasonable but we must take care about couple of
-     * things>
-     *  - Do not change image in the middle of transition. It messes up the view.
-     *  - Do not transition for the views which don't require it. Starting transition from
-     *      multiple fragments does weird things
-     *  - Do not wait to transition until the image loads from network
-     *
-     * Preview, cached image, network image, x - failed, o - succeeded
-     * P C N - start transition after...
-     * x x x - the cache fails
-     * x x o - the cache fails
-     * x o o - the cache succeeds
-     * o x o - the preview succeeds. Do not start on cache.
-     * o o o - the preview succeeds. Do not start on cache.
-     *
-     * So start transition after the first success or after anything with the cache
-     *
-     * @param isCacheRequest - is this listener for request image from cache or from the network
-     */
-    private inner class ImageRequestListener(
-            private val isCacheRequest: Boolean,
-            private val isThumnailRequest: Boolean) : RequestListener<Drawable> {
-
-        override fun onLoadFailed(e: GlideException?, model: Any, target: Target<Drawable>,
-                                  isFirstResource: Boolean): Boolean {
-            // If cache for full image failed complete transition
-            if (isCacheRequest && !isThumnailRequest && shouldStartTransition
-                    && !startedTransition) {
-                photoActionsListener.onBringUp()
```

Comment:
Without this code it doesn't start transition immediately on cache hit/ miss, it waits for the full download before unblocking the UI. I just tried it out and it's indeed the case. That's no-no.
I guess we could manually do this part with Glide (so do cache only request) and then do full download.